staging/src/k8s.io/kubectl/pkg/util/completion/completion_test.go

```diff
@@ -354,6 +354,62 @@ func TestPodResourceNameAndContainerCompletionFuncJointFormTooManyArgs(t *testin
 	checkCompletion(t, comps, []string{}, directive, cobra.ShellCompDirectiveNoFileComp)
 }
 
+func TestResourceAndPortCompletionFunc(t *testing.T) {
+	testCases := []struct {
+		name              string
+		args              []string
+		toComplete        string
+		expectedComps     []string
+		expectedDirective cobra.ShellCompDirective
+	}{
+		{
+			name:              "no args pod name",
+			args:              []string{},
+			toComplete:        "b",
+			expectedComps:     []string{"bar"},
+			expectedDirective: cobra.ShellCompDirectiveNoFileComp,
+		},
+		{
+			name:              "no args resources",
+			args:              []string{},
+			toComplete:        "s",
+			expectedComps:     []string{"services/", "statefulsets/"},
+			expectedDirective: cobra.ShellCompDirectiveNoFileComp | cobra.ShellCompDirectiveNoSpace,
+		},
+		{
+			name:              "too many args",
+			args:              []string{"pod-name", "port-number"},
+			toComplete:        "",
+			expectedComps:     []string{},
+			expectedDirective: cobra.ShellCompDirectiveNoFileComp,
+		},
+		{
+			name:              "joint from no args",
+			args:              []string{},
+			toComplete:        "pod/b",
+			expectedComps:     []string{"pod/bar"},
+			expectedDirective: cobra.ShellCompDirectiveNoFileComp,
+		},
+		{
+			name:              "joint from too many args",
+			args:              []string{"pod/pod-name", "port-number"},
+			toComplete:        "",
+			expectedComps:     []string{},
+			expectedDirective: cobra.ShellCompDirectiveNoFileComp,
+		},
```

Comment:
It seems you have completion with ports, is there a way to add tests for that too?

Reply:
+1 for this.  It would be nice to see a test that has something like this to test that it is able to get ports for pods and services:

```
expectedComps:     []string{"81:81", "82:82"},
```